lighthouse-extension/app/src/lighthouse-background.js

```diff
@@ -251,41 +257,56 @@ window.getDefaultAggregations = function() {
 
 /**
  * Save currently selected set of aggregation categories to local storage.
- * @param {!Array<{name: string, audits: !Array<string>}>} selectedAggregations
+ * @param {!{selectedAggregations: Array<{name: string, audits: !Array<string>}>,
+ *           disableExtensions: boolean}} settings
  */
-window.saveSelectedAggregations = function(selectedAggregations) {
+window.saveSettings = function(settings) {
   const storage = {
-    [STORAGE_KEY]: {}
+    [STORAGE_KEY]: {},
+    [SETTINGS_KEY]: {}
   };
 
+  // Stash selected aggregations.
   window.getDefaultAggregations().forEach(audit => {
-    storage[STORAGE_KEY][audit.name] = selectedAggregations.includes(audit.name);
+    storage[STORAGE_KEY][audit.name] = settings.selectedAggregations.includes(audit.name);
   });
 
+  // Stash disable extensionS setting.
+  disableExtensionsDuringRun = settings.disableExtensions;
+  storage[SETTINGS_KEY].disableExtensions = disableExtensionsDuringRun;
+
+  // Save object to chrome local storage.
   chrome.storage.local.set(storage);
 };
 
 /**
  * Load selected aggregation categories from local storage.
- * @return {!Promise<!Object<boolean>>}
+ * @return {!Promise<{selectedAggregations: Array<{name: string, audits: !Array<string>}>,
```

Comment:
this one was right, so `@return {!Promise<{selectedAggregations: !Object<boolean>, disableExtensions: boolean}>}`